### ISSUE TITLE
Why pred_logits are different in training, inference and paper?

### ISSUE THREAD

**Issue by yxchng** [mlpc-ucsd/MasQCLIP#6]
Training:

https://github.com/mlpc-ucsd/MasQCLIP/blob/2c21df65385b7a5f7ee586919116c0db8cf3a515/masqclip/masq_tuning.py#L229-L231

Inference:

https://github.com/mlpc-ucsd/MasQCLIP/blob/2c21df65385b7a5f7ee586919116c0db8cf3a515/masqclip/masq_tuning.py#L222-L223
https://github.com/mlpc-ucsd/MasQCLIP/blob/2c21df65385b7a5f7ee586919116c0db8cf3a515/masqclip/masq_tuning.py#L258-L259

Paper:

p_cls = p_obj * p_score

![Screenshot from 2023-12-04 11-52-08](https://github.com/mlpc-ucsd/MasQCLIP/assets/10518587/cb36dc9c-ef66-4eb2-b127-44aa38a9f5f0)

**Comment by xuxalan** [comment]
Thanks for your interest. For clarification, `outputs["pred_logits"]` comes from the mask classification module and corresponds to the logarithm of $p_{\text{clip}}^{(i)}$, and `mask_scores` represents the (logarithmic) scores of $p_{\text{obj}}$ derived from the mask proposal network. Since they are in logarithmic forms, we then add them (Line 258), which is the same with the paper.
In addition, scores are different during training and testing. As mentioned in the paper, $p_{\text{clip}}$ and $p_{\text{obj}}$ are supervised in two stages separately, and are combined during inference.

**Comment by yxchng** [comment]
why take score mask_scores[:, :, [0]] instead of mask_scores[:, :, [1]]? Isn't 1 positive prediction?

**Comment by xuxalan** [comment]
No, the first index corresponds to the positive prediction.
